src/bot.rs

```diff
@@ -220,7 +221,82 @@ impl Bot {
                                 info!("updating tester contacts");
                                 db::set_testers(conn, &filtered.collect::<Vec<_>>()).await?;
                             }
-                            // TODO: handle membership changes in review and submit group
+                            ChatType::Submit => {
+                                let new_members = filtered.collect::<HashSet<_>>();
+                                if new_members.len() == 1 {
+                                    info!("submit chat has only one member left, deleting");
+                                    db::delete_submit_chat(
+                                        &mut *state.db.acquire().await?,
+                                        chat_id,
+                                    )
+                                    .await?;
+                                    chat_id.delete(context).await?;
+                                }
+                            }
+                            ChatType::Review => {
+                                let mut conn = state.db.acquire().await?;
+                                let review_chat = db::get_review_chat(&mut conn, chat_id).await?;
+                                let new_members = filtered.collect::<HashSet<_>>();
+                                let old_members: HashSet<_> =
+                                    review_chat.get_members().into_iter().collect();
+                                let removed_members = old_members.difference(&new_members);
+                                for removed_member in removed_members {
+                                    // handle publisher removal
+                                    if *removed_member == review_chat.publisher {
+                                        match db::get_new_random_publisher(
+                                            &mut conn,
+                                            *removed_member,
+                                        )
+                                        .await
+                                        {
+                                            Ok(new_publisher) => {
+                                                db::set_review_chat_publisher(
+                                                    &mut conn,
+                                                    chat_id,
+                                                    new_publisher,
+                                                )
+                                                .await?;
+                                            }
+                                            Err(_) => warn!(
+                                                "Could not find a new publisher for chat {}",
+                                                chat_id
```

Comment:
better move this part to a separate function?

Reply:
I personally don't bother about it too much event though it's a lot of code because it's purpose is clear from the surrounding and it is not used anywhere else.